[2025-03-12T03:45:50.497+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: process_bronze_and_silver_layer.check_hourly_data_freshness manual__2025-03-12T03:45:30+00:00 [queued]>
[2025-03-12T03:45:50.501+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: process_bronze_and_silver_layer.check_hourly_data_freshness manual__2025-03-12T03:45:30+00:00 [queued]>
[2025-03-12T03:45:50.501+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-03-12T03:45:50.643+0000] {taskinstance.py:1382} INFO - Executing <Task(_PythonDecoratedOperator): check_hourly_data_freshness> on 2025-03-12 03:45:30+00:00
[2025-03-12T03:45:50.648+0000] {standard_task_runner.py:57} INFO - Started process 12290 to run task
[2025-03-12T03:45:50.651+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'process_bronze_and_silver_layer', 'check_hourly_data_freshness', 'manual__2025-03-12T03:45:30+00:00', '--job-id', '1051', '--raw', '--subdir', 'DAGS_FOLDER/5mins_processing_pipeline.py', '--cfg-path', '/tmp/tmp8d9upixo']
[2025-03-12T03:45:50.660+0000] {standard_task_runner.py:85} INFO - Job 1051: Subtask check_hourly_data_freshness
[2025-03-12T03:45:50.692+0000] {task_command.py:416} INFO - Running <TaskInstance: process_bronze_and_silver_layer.check_hourly_data_freshness manual__2025-03-12T03:45:30+00:00 [running]> on host c8e526380ef3
[2025-03-12T03:45:50.725+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='Andrew Tran' AIRFLOW_CTX_DAG_ID='process_bronze_and_silver_layer' AIRFLOW_CTX_TASK_ID='check_hourly_data_freshness' AIRFLOW_CTX_EXECUTION_DATE='2025-03-12T03:45:30+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-12T03:45:30+00:00'
[2025-03-12T03:45:50.730+0000] {5mins_processing_pipeline.py:128} INFO - Getting latest updated_at from bronze_layer.bronze_spins_hourly
[2025-03-12T03:45:50.730+0000] {5mins_processing_pipeline.py:101} INFO - Executing query: SELECT MAX(updated_at) FROM bronze_layer.bronze_spins_hourly
[2025-03-12T03:45:50.733+0000] {base.py:73} INFO - Using connection ID 'postgres_warehouse_db' for task execution.
[2025-03-12T03:45:50.738+0000] {warnings.py:109} WARNING - /opt/***/dags/5mins_processing_pipeline.py:104: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  df = pd.read_sql(query, conn)

[2025-03-12T03:45:50.741+0000] {5mins_processing_pipeline.py:105} INFO - Query executed successfully. Retrieved 1 rows.
[2025-03-12T03:45:50.742+0000] {5mins_processing_pipeline.py:136} INFO - Latest updated_at: 2025-03-11 17:53:01.567076
[2025-03-12T03:45:50.742+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 221, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/5mins_processing_pipeline.py", line 374, in check_hourly_data_freshness
    needs_refresh = latest_hourly_update < freshness_threshold
TypeError: '<' not supported between instances of 'str' and 'datetime.datetime'
[2025-03-12T03:45:50.745+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=process_bronze_and_silver_layer, task_id=check_hourly_data_freshness, execution_date=20250312T034530, start_date=20250312T034550, end_date=20250312T034550
[2025-03-12T03:45:50.751+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 1051 for task check_hourly_data_freshness ('<' not supported between instances of 'str' and 'datetime.datetime'; 12290)
[2025-03-12T03:45:50.795+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-03-12T03:45:50.810+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
